### ISSUE TITLE
[Caffe2] Output graph name is not preserved after compilation.

### ISSUE THREAD

**Issue by mciprian13** [pytorch/glow#3774]
For ONNX models all looks clean:
- the input/output placholder names are preserved after compilation (previously the output placeholders where prepended with "save_" which was ugly but now all is good).

For Caffe2 models the output placeholder name after compilation is not preserved:
- for example, for the LenetMnist bundle example, the original output placeholder name is "softmax"
whereas in the compiled model the name is "softmax__1" which is weird

In terms of usability and clarity, the output placeholder names for Caffe2 models should be preserved as well.

**Comment by nickgg** [comment]
Hi @mciprian13, this one looks like my fault.

It's due to uniqueness rules being enforced with Glow - previously the output `SaveNode` and the Placeholder itself had the same name, but this is not allowed inside Glow. It's odd that the Caffe2 and ONNX loaders behave differently, I'll investigate that.

I don't think it's possible to guarantee preservation of the output name, as I think we have stricter uniqueness guarantees than Caffe2 protos.

**Comment by mciprian13** [comment]
I understand Glow might be more strict than the proto but Glow should try to maintain the placeholder names in the graph since it directly impacts the bundle API and the user experience.
If the output placeholder is "softmax" the user expects to find the symbol "softmax" in the bundle API. It might come with a surprise and a headache if the symbol name is actually "softmax__1" (especially in the dynamic bundle API where the user application has to search for the symbol at run-time by parsing the symbol table).

**Comment by nickgg** [comment]
You're right, I think we should try to fix this. It's on my list.

**Comment by nickgg** [comment]
@mciprian13 can you please check if #3798 fixes your issue.

**Comment by mciprian13** [comment]
@nickgg I tested it on a couple of Caffe2 models and it works as expected. Thanks a lot for doing this!
